src/Processors/Transforms/FilterTransform.cpp

```diff
@@ -267,11 +264,9 @@ void FilterTransform::writeIntoQueryConditionCache(const MarkRangesInfoPtr & mar
         if (!buffered_mark_ranges_info)
             return;
 
-        query_condition_cache->write(
+        query_condition_cache_writer->write(
```

Comment:
Now that we have QueryConditionCacheWriter, I wondered if we can simplify all code in `writeIntoQueryConditionCache` (l. 255)?